apps/meteor/app/livechat/imports/server/rest/inquiries.ts

```diff
@@ -54,23 +52,20 @@ API.v1.addRoute(
 	'livechat/inquiries.take',
 	{ authRequired: true },
 	{
-		post() {
-			try {
-				check(this.bodyParams, {
-					inquiryId: String,
-					userId: Match.Maybe(String),
-				});
-				if (this.bodyParams.userId && !Users.findOneById(this.bodyParams.userId, { fields: { _id: 1 } })) {
-					return API.v1.failure('The user is invalid');
-				}
-				return API.v1.success({
-					inquiry: Meteor.runAsUser(this.bodyParams.userId || this.userId, () =>
-						Meteor.call('livechat:takeInquiry', this.bodyParams.inquiryId),
-					),
-				});
-			} catch (e) {
-				return API.v1.failure(e);
+		async post() {
+			check(this.bodyParams, {
+				userId: Match.Optional(String),
+				inquiryId: String,
+			});
+
+			const { userId, inquiryId } = this.bodyParams;
+
+			if (userId && !(await Users.findOneById(userId, { fields: { _id: 1 } }))) {
```

Comment:
`Users` is raw now, i believe it will ignore the `fields` key (use `projection` if possible)